### PR TITLE
expressions of dissatisfaction – models and rest interface

### PR DESCRIPTION
**Pull request by ushkarev**
depends on [cla_frontend#339](https://github.com/ministryofjustice/cla_frontend/pull/549)

### CODE REVIEW

**Comment by ushkarev** [comment]
test this please

File: cla_backend/apps/legalaid/models.py [hunk 1]
```diff
@@ -687,7 +718,7 @@ def split(self, user, category, matter_type1, matter_type2, assignment_internal)
                     'callback_attempt', 'search_field', 'provider_assigned_at'
                 ],
                 'clone_fks': [
-                    'thirdparty_details', 'adaptation_details'
+                    'thirdparty_details', 'adaptation_details', 'eod_details',
```
Comment (s-block) [inline cla_backend/apps/legalaid/models.py]: probably shouldn't be cloned when case is copied so can add 'eod_details' to 'excludes' = {... above and remove it from 'clone_fks' = {...

File: cla_backend/apps/legalaid/models.py [hunk 2]
```diff
@@ -205,6 +205,37 @@ class AdaptationDetails(CloneModelMixin, TimeStampedModel):
     }
 
 
+class EODDetails(CloneModelMixin, TimeStampedModel):
```
Comment (s-block) [inline cla_backend/apps/legalaid/models.py]: Probably shouldn't be cloned

File: cla_backend/apps/legalaid/models.py [hunk 3]
```diff
@@ -205,6 +205,37 @@ class AdaptationDetails(CloneModelMixin, TimeStampedModel):
...
+        instance = cls.objects.get(pk=pk)
+        copied_categories = [EODDetailsCategory(eod_details_id=eod_details.id,
+                                                category=category.category,
+                                                is_major=category.is_major) for category in instance.categories.all()]
+        eod_details.categories = copied_categories
+
+        return eod_details
+
+
+class EODDetailsCategory(CloneModelMixin, models.Model):
```
Comment (s-block) [inline cla_backend/apps/legalaid/models.py]: Probably shouldn't be cloned

File: cla_backend/apps/legalaid/models.py [hunk 4]
```diff
@@ -205,6 +205,37 @@ class AdaptationDetails(CloneModelMixin, TimeStampedModel):
...
+        eod_details = super(EODDetails, cls).clone_from_obj(pk, config)
+
+        # deep-copy EOD categories
+        if not pk or not eod_details:
+            return None
+        instance = cls.objects.get(pk=pk)
+        copied_categories = [EODDetailsCategory(eod_details_id=eod_details.id,
+                                                category=category.category,
+                                                is_major=category.is_major) for category in instance.categories.all()]
+        eod_details.categories = copied_categories
```
Comment (s-block) [inline cla_backend/apps/legalaid/models.py]: might be better as:

```python
        def clone_category(category):
            return EODDetailsCategory(
                eod_details_id=eod_details.id,
                category=category.category,
                is_major=category.is_major)

        eod_details.categories = map(clone_category, instance.categories.all())
```

**Pull request merged**